packages/contracts-bedrock/src/L2/L2ToL2CrossDomainMessenger.sol

```diff
@@ -204,7 +201,7 @@ contract L2ToL2CrossDomainMessenger is IL2ToL2CrossDomainMessenger, ISemver, Tra
             successfulMessages[messageHash] = true;
             emit RelayedMessage(source, nonce, messageHash);
         } else {
-            emit FailedRelayedMessage(source, nonce, messageHash);
+            revert TargetCallFailed();
         }
 
```

Comment:
Generally in favor of this. One small readability nit:

```solidity
        if (!success) {
            revert TargetCallFailed();
        }
        successfulMessages[messageHash] = true;
        emit RelayedMessage(source, nonce, messageHash);
```

Reply:
updated!